### PR TITLE
Search functionality is working

### LINKED ISSUE #16: Add item search API endpoint
/api/search will take a query string and return an array of results

### PR DESCRIPTION
**Pull request by theenderweggin**
Search bar now takes 10 item from amazon and displays them on the main page.  Image URL are not loading that will be an added ticket.  

closes #16

### CODE REVIEW

File: client/src/components/Search.jsx [hunk 1]
```diff
@@ -1,27 +1,54 @@
...
+  handleChange(event) {
+    this.setState({value: event.target.value});
+    console.log(this.state.value);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    // creates an action SEARCH_AMAZON and displatching that action to the reducer
+    var query = this.state.value;
+    store.dispatch(actions.searchAmazon(this.state.value));
```
Comment (iMobs) [inline client/src/components/Search.jsx]: Why are you using dispatch here as opposed to wrapping the component with connect to bind the actionCreator as a callback?

File: client/src/components/Search.jsx [hunk 2]
```diff
@@ -1,27 +1,54 @@
...
+  }
+  render() {
+    return (
+      <Row>
+        <Col md={12}>
+          <form onSubmit={this.handleSubmit}>
+            <input
               type="text"
+              className="searchBar"
+              value={this.state.textbox}
```
Comment (iMobs) [inline client/src/components/Search.jsx]: You use `this.state.value` everywhere else but here it's `this.state.textbox`?

File: client/src/store/reducers/results.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 export default (state = [], action) => {
   switch (action.type) {
+  case 'SEARCH_AMAZON':
```
Comment (iMobs) [inline client/src/store/reducers/results.js]: What's the point of this reducer if the query results are also being stored in `searchedItems`?

File: server/routes/auth.js [hunk 4]
```diff
@@ -9,7 +9,8 @@ router.route(['/', '/profile'])
     var state = {
       user: req.user, // get the user out of session and pass to template
       results: dummyData,
-
+      query: '',
```
Comment (iMobs) [inline server/routes/auth.js]: Where are these being used?

File: client/src/components/Search.jsx (hunk 2 above)
Comment (theenderweggin) [reply client/src/components/Search.jsx]: Great catch! I missed this one. 

Thanks

File: client/src/components/Search.jsx (hunk 1 above)
Comment (theenderweggin) [reply client/src/components/Search.jsx]: Not, sure why I used it.  It when I did my research I was instructed to do it this way. If you know a better one I'd be glad to learn it and change it but I don't know the why.

File: server/routes/auth.js (hunk 4 above)
Comment (theenderweggin) [reply server/routes/auth.js]: query isn't being used yet, its a future feature i want to add. will remove

**Pull request merged**